File: bfs_dfs_problems/bfs_dfs_problems__judge_reach.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;
using Tpl = System.Tuple<int, int>;

internal class Program
{
    static void Main()
    {
        int[] hw = ReadLine().Split().Select( int.Parse ).ToArray();
        int[] syx = ReadLine().Split().Select( int.Parse ).ToArray();
        int[] gyx = ReadLine().Split().Select( int.Parse ).ToArray();
        int[] dx = { 0, 0, -1, 1 };
        int[] dy = { -1, 1, 0, 0 };
        char[][] g = new char[hw[0]][];
        var q = new System.Collections.Generic.Queue<Tpl>();

        foreach( int i in Range( 0, hw[0] ) ) {
            g[i] = ReadLine().ToCharArray();
        }
        g[syx[0]][syx[1]] = '*';
        q.Enqueue( new Tpl( syx[0], syx[1] ) );
        while( 0 < q.Count ) {
            Tpl t = q.Dequeue();

            foreach( int i in Range( 0, 4 ) ) {
                int x = t.Item2 + dx[i];
                int y = t.Item1 + dy[i];

                if( x == gyx[1] && y == gyx[0] ) {
                    WriteLine( "Yes" );
                    return;
                }
                if( 0 <= x && x < hw[1] && 0 <= y && y < hw[0] && g[y][x] == '.' ) {
                    g[y][x] = '*';
                    q.Enqueue( new Tpl( y, x ) );
                }
            }
        }
        WriteLine( "No" );
        return;
    }
}

File: bfs_dfs_problems/bfs_dfs_problems__judge_reach.py
# 実行時間 10ms
from collections import deque

def main():
    h, w = list(map(int, input().split()))
    sy, sx = list(map(int, input().split()))
    gy, gx = list(map(int, input().split()))
    dx = [0, 0, -1, 1]
    dy = [-1, 1, 0, 0]
    g = [list(input()) for _ in [0] * h]
    q = deque()

    g[sy][sx] = "*"
    q.append((sy, sx))
    while 0 < len(q):
        t = q.popleft()

        for i in range(4):
            x = t[1] + dx[i]
            y = t[0] + dy[i]
            if x == gx and y == gy:
                return True
            if 0 <= x < w and 0 <= y < h and g[y][x] == ".":
                g[y][x] = "*"
                q.append((y, x))
    return False

print("Yes" if main() else "No")
